src/mempool/verify_tx.rs

```diff
@@ -501,16 +521,13 @@ pub mod test {
         let size = LaneBytesSize(data_proposal.estimate_size() as u64);
         let hash = data_proposal.hashed();
 
-        let signed_msg = ctx.mempool.crypto.sign(MempoolNetMessage::DataProposal(
-            hash.clone(),
-            data_proposal.clone(),
-        ))?;
+        let signed_msg = sign_msg_with_header(
+            &ctx.mempool.crypto,
+            MempoolNetMessage::DataProposal(hash.clone(), data_proposal.clone()),
+        )?;
```

Comment:
imho, I would prefer keep the same signature as before but update the `sign` function, instead of adding this "sign_msg_with_header".

If we still need the previous `sign` function, let's just move `sign_msg_with_header` to `BsltCrypto::sign_with_header`

Reply:
The reason I did this is because `sign` returns a SignedByValidator, while this needs a `MsgWithHeader`(which as network-specific logic).

Reply:
Arguably it's fine to have that in crypto still though